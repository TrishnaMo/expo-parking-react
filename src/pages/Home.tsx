import React, { useState } from "react";
import { Container, TextField, Button } from "@material-ui/core";
import { useHistory } from "react-router-dom";

const Home = () => {
  const history = useHistory();
  const [lotsCount, setLotsCount] = useState<string>("");

  const navigateToParkingLots = () => {
    const tempLots = parseInt(lotsCount);
    if (isNaN(tempLots) || tempLots <= 0) {
      alert("Please enter valid number");
    } else {
      history.push(`/parking-lots/${tempLots}`);
    }
  };

  return (
    <Container
      component="main"
      style={{
        display: "flex",
        height: "100vh",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      <Container
        maxWidth="xs"
        style={{
          border: "1px solid #ccc",
          padding: 30,
        }}
      >
        <TextField
          variant="outlined"
          fullWidth
          placeholder="Enter number of lots"
          onChange={(e) => setLotsCount(e.target.value)}
          value={lotsCount}
        />
        <Button
          color="primary"
          variant="contained"
          fullWidth
          style={{
            padding: 12,
            marginTop: 30,
          }}
          onClick={navigateToParkingLots}
          disabled={lotsCount.length === 0}
        >
          Submit
        </Button>
      </Container>
    </Container>
  );
};

export default Home;
