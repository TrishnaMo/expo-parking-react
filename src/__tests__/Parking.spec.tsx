import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "../pages/Home";

describe("Home page test cases", () => {
  it("Render snapshot to be expected", () => {
    expect(<Home />).toMatchSnapshot();
  });
  it("should show a button", () => {
    render(<Home />);
    expect(screen.getByRole("button", { name: "Submit" }));
  });
});
