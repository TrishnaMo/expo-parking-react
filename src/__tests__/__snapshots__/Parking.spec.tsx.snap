// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test cases Render snapshot to be expected 1`] = `<Home />`;
